src/lib/network/network_autoyast.rb

```diff
@@ -242,27 +274,25 @@ def assign_udevs_to_devs(udev_rules)
         next if !valid_rename_udev_rule?(rule)
         key.downcase!
 
+        # find item which matches to the given rule definition
         item, matching_item = LanItems.Items.find do |_, i|
           i["hwinfo"]["busid"].downcase == key || i["hwinfo"]["mac"].downcase == key
         end
         next if !matching_item
 
-        # for logging only
         name_from = matching_item["ifcfg"] || matching_item["dev_name"]
         log.info("Matching device found - renaming <#{name_from}> -> <#{name_to}>")
 
-        # selecting according device name is unreliable (selects only in between configured devices)
-        LanItems.current = item
+        # find rule in collision
+        colliding_item, _item_map = LanItems.Items.find do |i, _|
+          LanItems.GetDeviceName(i) == name_to
+        end
 
-        # find out what attribude is currently used for setting device name and
-        # change it if needed. Currently mac is used by default. So, we check is it is
-        # the other one (busid). If no we defaults to mac.
-        bus_attr = LanItems.GetItemUdev("KERNELS")
-        current_attr = bus_attr.empty? ? "ATTR{address}" : "KERNELS"
+        # rename item in collision
+        rename_lan_item(colliding_item, name_from)
```

Comment:
But omitting the attr and key arguments means that ReplaceItemUdev is not called, so we still end up with duplicate names.
And THAT is the thing we should have a test for.
The current tests just test the complicated API which only exists because `rename_lan_item` seems to serve two different purposes.